carma_wm/include/carma_wm/WorldModel.h

```diff
@@ -338,6 +351,42 @@ class WorldModel
    * \return map version
    */ 
   virtual size_t getMapVersion() const = 0;
+
+  /**
+   * \brief Overload: Returns the nearest point to the provided vehicle pose in the provided list
+   * 
+   * \param points The points and speed pairs to evaluate
+   * \param state The current vehicle state
+   * 
+   * \return index of nearest point in points
+   */
+  virtual int get_nearest_point_index(const std::vector<PointSpeedPair>& points,
```

Comment:
I think we need a different name for this function. It is not clear to a user why this function uses downtrack distance instead of cartesian distance. Additionally, the comments should be updated to provide more detail about the implementation, why they might be used, and high cost of these methods. 

It seems Anish has them in basic autonomy as well. Should we move these to that library?

Reply:
Moved these functions to basic_autonomy library